test/e2e/ethereum_contract_test.go

```diff
@@ -288,23 +301,27 @@ func (suite *EthereumContractTestSuite) TestFFIInvokeMethod() {
 	invokeContractRequest := &fftypes.InvokeContractRequest{
 		Location: locationBytes,
 		Params: map[string]interface{}{
-			"newValue": float64(42),
+			"newValue": float64(3),
 		},
 	}
 
 	<-received1
-	<-changes1 // also expect database change events
 
-	res, err := InvokeFFIMethod(suite.testState.t, suite.testState.client1, suite.interfaceID, ffi.Methods[0].Name, invokeContractRequest)
+	res, err := InvokeFFIMethod(suite.testState.t, suite.testState.client1, suite.contractID, ffi.Methods[0].Name, invokeContractRequest)
 	assert.NoError(suite.testState.t, err)
 	assert.NotNil(suite.testState.t, res)
 
-	<-received1
-	<-received1
+	match := map[string]interface{}{
+		"info": map[string]interface{}{
+			"address": suite.contractAddress,
+		},
+		"outputs": map[string]interface{}{
+			"_value": "3",
+			"_from":  suite.testState.org1.Identity,
+		},
+		"subscription": sub.ID.String(),
+	}
 
-	events := GetContractEvents(suite.T(), suite.testState.client1, suite.testState.startTime)
-	assert.Equal(suite.T(), 1, len(events))
-	assert.Equal(suite.T(), "Changed", events[0].Name)
-	assert.Equal(suite.T(), "42", events[0].Outputs.GetString("_value"))
-	assert.Equal(suite.T(), suite.ethIdentity, events[0].Outputs.GetString("_from"))
+	event := waitForChangeEvent(suite.T(), suite.testState.client1, changes1, match)
```

Comment:
Does this only validate the `changes1` websocket for database events? Shouldn't we drain the `received1` websocket as well (as that's the more important user-facing one for E2E purposes)?

Reply:
Do you think we should wait for both here, or just look at `received1`?

Reply:
Okay, updated this PR to look at the deliveryEvents